LogMonitor/src/LogMonitor/LogFileMonitor.cpp

```diff
@@ -1683,13 +1672,38 @@ LogFileMonitor::ReadLogFile(
 }
 
 void LogFileMonitor::WriteToConsole( _In_ std::wstring Message, _In_ std::wstring FileName) {
-    wstring prefix;
-    if (m_includeFileNames)
-    {
-        prefix = Utility::FormatString(L"[Log File: %s] ", FileName.c_str());
-    }
+    auto logFmt = L"{\"Source\":\"File\",\"LogEntry\":{\"Logline\":\"%s\",\"FileName\":\"%s\"},\"SchemaVersion\":\"1.0.0\"}";
+    size_t start = 0;
+    size_t i = 0;
+    wstring msg;
+
+    while (true) {
+        i = Message.find(L"\n", start);
+        if (i == std::string::npos) {
+            // only one-line log or remaining line without \n
+            msg = Message.substr(start, Message.size() - start);
+        }
+        else {
+            // substr except ith (\n)
+            msg = Message.substr(start, i - start);
+            start = i + 1;
+            // remove \r if any, usually before \n
+            if (msg.substr(msg.size() - 1) == L"\r") {
+                msg.replace(msg.size() - 1, 1, L"");
+            }
+        }
+        // ignore empty lines
+        if (msg.size() > 0) {
+            // escape backslashes in FileName
+            auto fmtFileName = Utility::ReplaceAll(FileName, L"\\", L"\\\\");
+            // sanitize msg
+            Utility::SanitizeJson(msg);
+            auto log = Utility::FormatString(logFmt, msg.c_str(), fmtFileName.c_str());
+            logWriter.WriteConsoleLog(log);
+        }
 
-    logWriter.WriteConsoleLog(prefix + Utility::ReplaceAll(Message, L"\n", L"\n" + prefix));
+        if (i >= Message.size()) break;

```

Comment:
@profnandaa what is going on here, the logic? and why is this required?

Reply:
This is where the actual write to console is taking place? yes?

Reply:
This is to make sure we break out of the while loop once we are at the end of the `Message` string. The actual printing is happening at L1702 above.